l10n_it_financial_statements_report/hooks.py

```diff
@@ -0,0 +1,132 @@
+#  Copyright 2023 Simone Rubino - TAKOBI
+#  License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+from openupgradelib.openupgrade import logged_query
+
+from odoo.tools import DotDict
+
+NEW_MODULE_NAME = "l10n_it_financial_statements_report"
+OLD_MODULE_NAME = "l10n_it_account_balance_report"
+
+REMOVED_MODELS = [
+    "report_trial_balance_account",
+    "account_balance_report_partner",
+    "account_balance_report_account",
+]
+
+RENAMED_MODELS = [
+    (
+        "account_balance_report",
+        "report.l10n_it_financial_statements_report.report",
+    ),
+    (
+        "report.l10n_it_a_b_r.account_balance_report_xlsx",
+        "report.l10n_it_financial_statements_report.report_xlsx",
+    ),
+]
+
+RENAMED_FIELDS = [
+    (
+        "account.account.type",
+        "account_account_type",
+        "account_balance_report_section",
+        "financial_statements_report_section",
+    ),
+]
+
+RENAMED_XMLIDS = [
+    (
+        "template_account_balance_report",
+        "template_financial_statements_report",
+    ),
+    (
+        "account_balance_report",
+        "report",
+    ),
+    (
+        "account_balance_report_base",
+        "financial_statements_report_base",
+    ),
+    (
+        "account_balance_report_title",
+        "financial_statements_report_title",
+    ),
+    (
+        "account_balance_report_filters",
+        "financial_statements_report_filters",
+    ),
+    (
+        "account_balance_report_split_columns",
+        "financial_statements_report_split_columns",
+    ),
+    (
+        "account_balance_report_lines_header",
+        "financial_statements_report_lines_header",
+    ),
+    (
+        "account_balance_report_lines",
+        "financial_statements_report_lines",
+    ),
+    (
+        "account_balance_report_partners",
+        "financial_statements_report_partners",
+    ),
+    (
+        "account_balance_report_totals",
+        "financial_statements_report_totals",
+    ),
+]
+
+
+def remove_models(cr, model_spec):
+    for name in model_spec:
+        logged_query(
+            cr,
+            "DELETE FROM ir_model WHERE model = %s",
+            (name,),
+        )
+
+
+def migrate_old_module(cr):
+    remove_models(
```

Comment:
no need, odoo does it automatically.

Reply:
When this method is executed in the `pre_init_hook`, without this change I have WARNING messages like:
> odoo.addons.base.models.ir_model: The model account_balance_report_account could not be dropped because it did not exist in the registry.

Reply:
Ok, sorry, I understand that some things have to be done different when you are calling migrations scripts in a pre-init-hook. That's something unusual so I am not sure which differences encompass this approach. Feel free!